### PR TITLE
Remove the node from dnsmasq config when shutting down

### PR DESCRIPTION
**Pull request by smarterclayton**
Prevents dnsmasq from pending a dead server when the network pod dies or is interrupted. Also prevents us serving DNS before our caches fill.

https://bugzilla.redhat.com/show_bug.cgi?id=1584995

### CODE REVIEW

**Comment by smarterclayton** [comment]
@liggitt

**Comment by smarterclayton** [comment]
Well, at least I know I broke DNS.

File: pkg/cmd/server/start/start_node.go [hunk 1]
```diff
@@ -115,14 +116,26 @@ var networkLong = templates.LongDesc(`
 
 // NewCommandStartNetwork provides a CLI handler for 'start network' command
 func NewCommandStartNetwork(basename string, out, errout io.Writer) (*cobra.Command, *NodeOptions) {
-	options := &NodeOptions{Output: out}
+	options := &NodeOptions{
+		ExpireDays: crypto.DefaultCertificateLifetimeInDays,
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply pkg/cmd/server/start/start_node.go]: You can't start openshift start network anymore without args - someone broke it in a refactor at some point in the last month or so.

File: pkg/cmd/server/start/start_node.go [hunk 2]
```diff
@@ -115,14 +116,26 @@ var networkLong = templates.LongDesc(`
...
 		Use:   "network",
 		Short: "Launch node network",
 		Long:  fmt.Sprintf(networkLong, basename),
 		Run: func(c *cobra.Command, args []string) {
-			options.Run(c, errout, args, wait.NeverStop)
+			ch := make(chan struct{})
+			interrupt.New(func(s os.Signal) {
+				close(ch)
+				fmt.Fprintf(errout, "interrupt: Gracefully shutting down ...\n")
+				time.Sleep(200 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply pkg/cmd/server/start/start_node.go]: 5 times better.

File: pkg/cmd/server/start/start_node.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (knobunc) [reply pkg/cmd/server/start/start_node.go]: Ugh :-(

**Comment by smarterclayton** [comment]
This is not ready to merge, there are failing tests.

**Comment by smarterclayton** [comment]
Ok, unit tests are fine.  @liggitt I need to make sure I pick a port number that won't get used in the near future on the kubelet.

**Comment by smarterclayton** [comment]
@liggitt moved it to 11256 and added a comment

**Comment by sdodson** [comment]
Nevermind, already on release-3.10, just waiting on an ocp build.

**Pull request merged**